packages/snaps-execution-environments/src/common/endowments/abortController.ts

```diff
@@ -0,0 +1,16 @@
+/**
```

Comment:
A lot of these files are the same, except for the object being hardened. Couldn't you use an array of all the objects that need to be hardened, and to the hardening in a loop somehow? That would save a bunch of code duplication.

Reply:
Well, yes, somehow. Applying hardening is pretty much easy thing to do. But, it has to do things with how we provide and manage these endowments in the `index.ts` where it's expected to have a factory.

So, I tried to make this consistent with the other (previously attenuated) endowments that had some kind of modification (fetch, timeout, interval). Also, the tests would require hardening first where each endowment is tested on some atomic level, which does not involve `index.ts` and procedures for dependency injection implemented there. That's why each endowment is hardened on a factory level.

I could make most of these into one file with all of these factories and then export/test them like that when needed (still not sure if it's better or just more confusing). But, still there will be some as separate like `network`, `interval`, `timeout`, `crypto` which contain some heavy attenuation even before hardening. So, it was mostly because of consistency and trying to make clean understandable testing flow.

This can be an open question if there are any proposals or strong opinions. I'm happy to follow and implement whatever seems to be better if it doesn't add more functional or testing complexity!

Reply:
Nothing forbids a factory factory

Reply:
And if that's where you go with it, you can have a single collection of endowments that you use for hardening and returning and also for testing them.

Reply:
To answer on this I've added endowment factory that builds consolidated endowment factories. There are still endowments that are attenuated in a custom way, like they were before. Others are hardened through the factory making process from a single collection. Only `TextDecoder` and `TextEncoder` are left as the separate modules since they're causing issues in the `snaps-controllers` because there is a problem with testing environments and availability of these endowments in `jest/jsdom`. At the end all of these common javascript endowments are provided by calling `buildCommonEndowments()` function where needed.